bot/exts/moderation/watchchannels/_watchchannel.py

```diff
@@ -353,8 +353,16 @@ async def prepare_watched_users_data(
             watched_iter = reversed(watched_iter)
 
         list_data["info"] = {}
+
+        # Pre-fetch all watched users to avoid sequential API calls
+        members = await asyncio.gather(*[
+            get_or_fetch_member(ctx.guild, user_id)
+            for user_id, _ in watched_iter
```

Comment:
Why is this making a possible ton of api requests? Additionally, gathering won't speed this up due to the library's implementation of ratelimit handling, internally each request is made sequentially.

Reply:
Because there is a large majority of users who are not in the bot cache (since they're off-guild) so we need to fetch their user.

Does you suggestion of using Guild.query_members fetch users that aren't in the guild?

Reply:
What do you mean off-guild? Does this mean they are no longer a member in the server?